controllers/main_test.go

```diff
@@ -445,3 +459,254 @@ func TestNewMainController(t *testing.T) {
+			}}, {
+			// CreateMultisig
+			err: errors.New("error"),
+		}},
+		queryArgs:     []driver.Value{46},
+		row:           tUser,
+		noSecondQuery: true,
+		wantErrCode:   codes.Unavailable,
+	}, {
+		name:           "unable to serialize redeem script",
+		userID:         46,
+		userPubKeyAddr: tUserPubKeyAddr,
+		stakepooldQueue: []queueItem{{
+			// ValidateAddress
+			thing: &pb.ValidateAddressResponse{
+				IsMine:     true,
+				PubKeyAddr: tPoolPubKeyAddr,
+			}}, {
+			// CreateMultisig
+			thing: &pb.CreateMultisigResponse{
+				RedeemScript: "bogus script",
+				Address:      tMSA,
+			}}},
+		queryArgs:     []driver.Value{46},
+		row:           tUser,
+		noSecondQuery: true,
+		wantErrCode:   codes.Unavailable,
+	}, {
+		name:           "unable to import redeem script",
+		userID:         46,
+		userPubKeyAddr: tUserPubKeyAddr,
+		stakepooldQueue: []queueItem{{
+			// ValidateAddress
+			thing: &pb.ValidateAddressResponse{
+				IsMine:     true,
+				PubKeyAddr: tPoolPubKeyAddr,
+			}}, {
+			// CreateMultisig
+			thing: &pb.CreateMultisigResponse{
+				RedeemScript: tRedeemScript,
+				Address:      tMSA,
+			}}, {
+			// ImportNewScript
+			err: errors.New("error"),
+		}},
+		queryArgs:     []driver.Value{46},
+		row:           tUser,
+		noSecondQuery: true,
+		wantErrCode:   codes.Unavailable,
+	}}
+	c := web.C{Env: map[interface{}]interface{}{}}
+	for _, test := range tests {
+		mock, dbMap := makeDB()
+		sm := tManagerWithQueue(test.stakepooldQueue)
+		cfg := &Config{StakepooldServers: sm, NetParams: params, VotingXpub: vXpub, FeeXpub: fXpub}
+		mc := &MainController{Cfg: cfg}
+		// Set expected database queries and updates.
+		if test.queryArgs != nil {
+			mock.ExpectQuery(`^SELECT (.*) FROM Users WHERE UserId = (.+)$`).
+				WithArgs(test.queryArgs...).
+				WillReturnRows(sqlmock.NewRows(tUserCol).AddRow(test.row...))
+			if !test.noSecondQuery {
+				mock.ExpectQuery(`^SELECT (.*) FROM Users WHERE UserId = (.+)$`).
+					WithArgs(test.queryArgs...).
+					WillReturnRows(sqlmock.NewRows(tUserCol).AddRow(test.row...))
+			}
+		}
+		if test.updateArgs != nil {
+			mock.ExpectExec("^update `Users` set `Email`=(.+)," +
+				" `Username`=(.+), `Password`=(.+), `MultiSigAddress`=(.+), " +
+				"`MultiSigScript`=(.+), `PoolPubKeyAddr`=(.+), `UserPubKeyAddr`=(.+), " +
+				"`UserFeeAddr`=(.+), `HeightRegistered`=(.+), `EmailVerified`=(.+), " +
+				"`EmailToken`=(.+), `APIToken`=(.+), `VoteBits`=(.+), " +
+				"`VoteBitsVersion`=(.+) where `UserId`=(.+);$").
+				WithArgs(test.updateArgs...).
+				WillReturnResult(sqlmock.NewResult(0, 0))
+		}
+		c.Env["APIUserID"] = test.userID
+		// nil APIUserID indicats an authentication error occured.
+		// Simulating that on userid -1.
+		if test.userID == -1 {
+			c.Env["APIUserID"] = nil
+		}
+		c.Env["DbMap"] = dbMap
+		// Put userPubKeyAddr as form value.
+		r, _ := http.NewRequest("GET", "?UserPubKeyAddr="+test.userPubKeyAddr, nil)
+		_, errCode, _, _ := mc.APIAddress(c, r)
```

Comment:
I wonder if we should also be checking the other values returned by `APIAddress()`.

This actually makes me wonder, do we really need to return both a string response and an error? The error alone is probably sufficient, having both seems redundant.